### PR TITLE
Fix constants missing notices

### PR DESCRIPTION
**Pull request by scottcwilson**
As reported in #2564.

### CODE REVIEW

File: admin/includes/modules/document_general/collect_info.php [hunk 1]
```diff
@@ -18,21 +18,21 @@
   'products_model' => '',
   'products_image' => '',
   'products_price' => '0.0000',
-  'products_virtual' => DEFAULT_DOCUMENT_GENERAL_PRODUCTS_VIRTUAL,
+  'products_virtual' => DEFAULT_DOCUMENT_PRODUCT_PRODUCTS_VIRTUAL,
```
Comment (drbyte) [inline admin/includes/modules/document_general/collect_info.php]: Should Document-General be using Document-Product defaults?

**Comment by scottcwilson** [comment]
It's a much bigger fix otherwise.  Are you sure it's worth it to change the DB and add the missing settings?

**Comment by drbyte** [comment]
> It's a much bigger fix otherwise. Are you sure it's worth it to change the DB and add the missing settings?

I'd prefer not to.

How much of an issue is it to share defaults between the 2? @lat9 ?

(If I could get away with it I'd delete all these "other" product-types. I never use them myself, but I know some people do.)

**Comment by scottcwilson** [comment]
@zcwilt mentioned he wanted to get rid of the music stuff.  Me too.  

I will do it if requested, just wanted to use this shortcut for now.

**Comment by scottcwilson** [comment]
BTW this has been a bug since 1.3.0 (if not before).  We're only seeing this now because of PHP 7.2+.

**Comment by scottcwilson** [comment]
I could just hardcode these to 0 if you prefer.

**Comment by lat9** [comment]
My feeling is that the best approach is to do as @scottcwilson indicated above.  Given that the processing has essentially been using empty values _forever_, just hardcode those settings to their 'off' positions.

**Comment by lat9** [comment]
> (If I could get away with it I'd delete all these "other" product-types. I never use them myself, but I know some people do.)
Oh, please don't!  I've got clients that have re-purposed those product types and the built-in alternates provide some basic guidance for plugin authors that are generating new product types.

**Comment by drbyte** [comment]
> I could just hardcode these to 0 if you prefer.

> just hardcode those settings to their 'off' positions.

Okay, yes, let's do that. It's actually the most sensible given what those entries are for.

**Pull request merged**